sockeye/data_io.py

```diff
@@ -1442,6 +1443,20 @@ def load(fname: str) -> 'ParallelDataSet':
             total_splits = horovod_mpi.hvd.size()
             i = split_index / total_splits
             j = (split_index + 1) / total_splits
+            # For each bucket, check if there are more splits (workers) than
+            # sentences.  If so, replicate that bucket's sentences N times where
+            # N is the minimum number required so that each split has at least
+            # one sentence.  This is not required for empty buckets since all
+            # splits will contain zero sentences.
+            for k in range(len(source)):
+                num_sentences = len(source[k])
+                if num_sentences > 0:
+                    num_copies = math.ceil(total_splits / num_sentences)
+                    if num_copies > 1:
+                        logger.info('Replicating bucket of %d sentence(s) %d times to cover %d splits.',
+                                    num_sentences, num_copies, total_splits)
+                        source[k] = mx.nd.concat(*[source[k] for _ in range(num_copies)], dim=0)
```

Comment:
could we use `mx.nd.repeat` here?
```suggestion
                        source[k] = source[k] = mx.nd.repeat(source[k], repeats=num_copies, axis=0)
```

Reply:
Good recommendation.